ecommerce/qverse_features/paystack/test_client.py

```diff
@@ -0,0 +1,269 @@
+        self.assertDictEqual(actual_return_dict, expected_return_dict)
+
+    def test_verify_transaction(self):
+        """
+        Verifies that client returns required data to send Paystack verify transcation request.
+        """
+        reference = 'fake_reference'
+        expected_return_dict = {
+            'method': 'GET',
+            'path': '/transaction/verify/{}'.format(reference)
+        }
+        actual_return_dict = self.client.verify_transaction(reference)
+        self.assertDictEqual(actual_return_dict, expected_return_dict)
+
+    def test_create_refund(self):
+        """
+        Verifies that client returns required data to send Paystack create refund request.
+        """
+        expected_return_dict = {
+            'method': 'POST',
+            'data': {
+                'transaction': self.TEST_DATA
+            },
+            'path': '/refund'
+        }
+        actual_return_dict = self.client.create_refund(self.TEST_DATA)
+        self.assertDictEqual(actual_return_dict, expected_return_dict)
+
+    def test_fetch_refund(self):
+        """
+        Verifies that client returns required data to send Paystack fetch refund request.
+        """
+        refund_id = 'fake_id'
+        expected_return_dict = {
+            'method': 'GET',
+            'path': '/refund/{}'.format(refund_id)
+        }
+        actual_return_dict = self.client.fetch_refund(refund_id)
+        self.assertDictEqual(actual_return_dict, expected_return_dict)
+
+    @patch('ecommerce.qverse_features.paystack.client.PaystackClient.handle_request')
+    def test_handler_for_verify_tarsanction(self, mock_handle_request):
+        """
+        Verifies that function calls correct client method to handle verify tarnsaction request.
+        """
+        expected_code = 'VERIFY_TRANSACTION'
+        expected_transaction_object = self.client.verify_transaction(self.TEST_DATA)
+        self.client.handler(expected_code, self.TEST_DATA)
+        mock_handle_request.assert_called_once_with(expected_transaction_object)
+
+    @patch('ecommerce.qverse_features.paystack.client.PaystackClient.handle_request')
+    def test_handler_for_initialize_tarsanction(self, mock_handle_request):
+        """
+        Verifies that function calls correct client method to handle initailize tarnsaction request.
+        """
+        expected_code = 'INITIALIZE_TRANSACTION'
+        expected_transaction_object = self.client.initialize_transaction(self.TEST_DATA)
+        self.client.handler(expected_code, self.TEST_DATA)
+        mock_handle_request.assert_called_once_with(expected_transaction_object)
+
+    @patch('ecommerce.qverse_features.paystack.client.PaystackClient.handle_request')
+    def test_handler_for_fetch_refund(self, mock_handle_request):
+        """
+        Verifies that function calls correct client method to handle fetch refund request.
+        """
+        expected_code = 'FETCH_REFUND'
```

Comment:
I see that you have made constants for these codes. Can we those constants here as well?

Reply:
I have intentionally used hardcoded string here instead of constant as in this way we are indirectly testing our constant value also.
And in the future, if someone changes constant value than they need to update the related test also.